### PR TITLE
GridDevice: Exclude MeasurementGates in validation of qubit pairs

### LINKED ISSUE #5652: GridDevice should allow measuring any qubit
**Description of the issue**

Currently `cirq_google.GridDevice` requires that for any operation with exactly two qubits, the pair of qubits must be in `qubit_pairs`. However, not all 2-qubit operations require the qubits to be coupled. In particular, `MeasurementGate` should be allowed on any set of qubits, including pairs that are not coupled.

**How to reproduce the issue**

```
In [1]: import cirq; import cirq_google

In [2]: cirq_google.Sycamore.validate_operation(cirq.measure(cirq.q(0, 5), cirq.q(2, 5), cirq.q(4, 5)))

In [3]: cirq_google.Sycamore.validate_operation(cirq.measure(cirq.q(0, 5), cirq.q(2, 5)))
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-5-a955df2ee85c> in <module>
----> 1 cirq_google.Sycamore.validate_operation(cirq.measure(cirq.q(0, 5), cirq.q(2, 5)))

~/src/quantumlib/cirq/cirq-google/cirq_google/devices/grid_device.py in validate_operation(self, operation)
    341             and frozenset(operation.qubits) not in self._metadata.qubit_pairs
    342         ):
--> 343             raise ValueError(f'Qubit pair is not valid on device: {operation.qubits!r}.')
    344 
    345     def __str__(self) -> str:

ValueError: Qubit pair is not valid on device: (cirq.GridQubit(0, 5), cirq.GridQubit(2, 5)).
```

**Cirq version**

0.15

### PR DESCRIPTION
**Pull request by verult**
I'm comfortable with special-casing MeasurementGate because it's the only gate today with the property that it can be applied to any subset of qubits.

Fixes #5652

@maffoo

### CODE REVIEW

File: cirq-google/cirq_google/devices/grid_device_test.py [hunk 1]
```diff
@@ -255,12 +255,27 @@ def test_grid_device_validate_operations_positive():
...
+    # Measurement across vertical qubit pairs (invalid pairs) should succeed.
+    for i in range(GRID_HEIGHT - 1):
+        device.validate_operation(
+            cirq.measure(device_info.grid_qubits[2 * i], device_info.grid_qubits[2 * (i + 1)])
+        )
+        device.validate_operation(
+            cirq.measure(
+                device_info.grid_qubits[2 * i + 1], device_info.grid_qubits[2 * (i + 1) + 1]
+            )
+        )
```
Comment (maffoo) [inline cirq-google/cirq_google/devices/grid_device_test.py:278]: Would be good to add some test cases of measurement on more qubits, for example triplets of qubits or all qubits (`cirq.measure(*device_info.grid_qubits)`).

**Review by maffoo — APPROVED** [review]

**Comment by dabacon** [comment]
What about identity gate, does it also need to be allowed?  (Yes odd to do identity across multiple qubits, but I think it is technically a multiqubit gate.  Reset isn't)

**Comment by verult** [comment]
Ah right. I left identity gate as a TODO which will be addressed soon in a separate PR. Will change the check to something like `not any(operation in gf for gf in SUBSET_GATE_FAMILIES)` in a separate PR

**Comment by dstrain115** [comment]
What about wait gate?  Does that need to be handled here?

**Comment by verult** [comment]
@dstrain115 does this mean only single-qubit WaitGates are allowed by the CircuitSerializer since both `num_qubits` and `qid_shape` are None in the WaitGate constructor?

https://github.com/quantumlib/Cirq/blob/b592ae79cd880174a55c67bbc8439335b108c816/cirq-google/cirq_google/serialization/circuit_serializer.py#L568

**Comment by dstrain115** [comment]
It's been a while since I dealt with this, but I think wait gates are similar to measurement gates in that they can be any permutation of qubits.  I think they were the PERMUTATION_SET in the old device spec.

**Comment by verult** [comment]
Added WaitGate support

File: cirq-google/cirq_google/devices/grid_device.py [hunk 2]
```diff
@@ -37,6 +37,9 @@
 MEASUREMENT_GATE_FAMILY = cirq.GateFamily(cirq.MeasurementGate)
 WAIT_GATE_FAMILY = cirq.GateFamily(cirq.WaitGate)
 
+# Families of gates which can be applied to any subset of valid qubits.
+_SUBSET_PERMUTATION_GATE_FAMILIES = [MEASUREMENT_GATE_FAMILY, WAIT_GATE_FAMILY]
```
Comment (maffoo) [inline cirq-google/cirq_google/devices/grid_device.py:41]: Could we perhaps call this `_VARIADIC_GATE_FAMILIES` instead? I think the term "variadic" captures the property we want, that these are gates that can be applied to any number of qubits.

**Review by maffoo — COMMENTED** [review]
I'd suggest using the term "variadic" to describe these gates, otherwise this looks good to me.

File: cirq-google/cirq_google/devices/grid_device.py (hunk 2 above)
Comment (dstrain115) [reply cirq-google/cirq_google/devices/grid_device.py:41]: I am going to remember this vocabulary word for when I take the Quantum SATs!

**Pull request merged**